rig/machine_control/machine_controller.py

```diff
@@ -1427,6 +1460,22 @@ def __init__(self, machine_controller, x, y, start_address, end_address):
         # Current offset from start address
         self._offset = 0
 
+    @property
+    def buffer_size(self):
+        """Return the number of bytes in the write buffer."""
+        return self._write_buffer.buffer_size
+
+    def close(self):
+        """Flush and close the file-like."""
+        if not self._closed:
+            self.flush()
+            self._closed = True
+
+    @property
```

Comment:
More that it's exposing a property of a internal object that it's not
expected the programmer will have knowledge of

On 1 May 2015 at 08:15, Jonathan Heathcote <notifications@github.com> wrote:

> In rig/machine_control/machine_controller.py
> <https://github.com/project-rig/rig/pull/125#discussion_r29492259>:
>
> > @@ -1427,6 +1460,22 @@ def __init__(self, machine_controller, x, y, start_address, end_address):
> >          # Current offset from start address
> >          self._offset = 0
> >
> > +    @property
> > +    def buffer_size(self):
> > +        """Return the number of bytes in the write buffer."""
> > +        return self._write_buffer.buffer_size
> > +
> > +    def close(self):
> > +        """Flush and close the file-like."""
> > +        if not self._closed:
> > +            self.flush()
> > +            self._closed = True
> > +
> > +    @property
>
> As in for keeping it read-only? Fair enough then :).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/project-rig/rig/pull/125/files#r29492259>.
>